src/common/lib/config_storage.c

```
@@ -363,94 +443,170 @@ uint8_t config_get_led_brightness(void) {
-    
+
     // Read old copy from flash (to preserve after erase)
-    const config_data_t *old_copy = read_flash_config(old_copy_index);
-    
+    const config_data_t* old_copy = read_flash_config(old_copy_index);
+
     // Copy old config to buffer (if valid, otherwise use current config with seq-1)
     if (validate_config(old_copy)) {
-        memcpy(sector_buffer + (old_copy_index * CONFIG_COPY_SIZE), 
-               old_copy, sizeof(config_data_t));
+        memcpy(sector_buffer + (old_copy_index * CONFIG_COPY_SIZE), old_copy,
```

Clang (14.0.6)

[warning] result of multiplication in type 'int' is used as a pointer offset after an implicit widening conversion to type 'ptrdiff_t'

(bugprone-implicit-widening-of-multiplication-result)